interpreter/text/parser.mly

```diff
@@ -850,6 +869,10 @@ module_fields1 :
     { fun c -> let mf = $2 c in
       fun () -> let m = mf () in
       {m with exports = $1 c :: m.exports} }
+  | exn module_fields
+    { fun c -> let mf = $2 c in
```

Comment:
You must invoke $1 c before $2 c, otherwise the bindings will not have been added before checking the rest. See e.g. the memory case.

Nit: Move up before globals case

Reply:
Thanks. That fixed one scoping issue, and exposed another. I will put a comment on the relevant place in a few moments.